schemas/README.md

```diff
@@ -1,17 +1,146 @@
 # Maven module containing schema plugins
 
-## Plugin structure
+## Schema Plugin Definition
 
-A schema plugin is composed off:
+* `src/main/plugin/<folder>` plugin contents:
+   
+  * `schema-ident.xml` configuration file
+  * schema definition `xsd` files
+  * transformations `xsl` files
+  * See [GeoNetwork Manual](https://geonetwork-opensource.org/manuals/trunk/en/customizing-application/implementing-a-schema-plugin.html)
 
-* A configuration folder containing XML file for configuration,
-the schema as XSDs and a set of transformation (See http://geonetwork-opensource.org/manuals/trunk/eng/developer/schemaPlugins/index.html).
+* `src/main/java` providing:
+  
+  * Optional: SchemaPlugin bean
+  * Optional: ApplicationListener<ServerStartup> to auto install plugin
 
-* (optional) A SchemaPlugin bean
+## Recommended: Artifact unpack
 
+1. Be sure your plugin includes `src/main/plugin`.
 
-## Add a plugin to the build
+2. Use `mvn install` to install your schema plugin in your local repository.
 
-To include a new schema plugin in a build, copy the schema folder
-here and add it to the copy-schemas execution in web/pom.xml.
+3. Add a profile to `web/pom.xml` unpacking your schema plugin.
+   
+   ```xml
+    <profile>
+      <id>hnap</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-dependency-plugin</artifactId>
+            <dependencies>
+              <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>schema-iso19139.ca.HNAP</artifactId>
+                <version>${gn.schemas.version>}</version>
+              </dependency>
+            </dependencies>
+            <executions>
+              <execution>
+                <id>iso19139.ca.HNAP-resources</id>
+                <phase>process-resources</phase>
+                <goals><goal>unpack</goal></goals>
+                <configuration>
+                  <encoding>UTF-8</encoding>
+                  <artifactItems>
+                    <artifactItem>
+                      <groupId>org.geonetwork-opensource</groupId>
+                      <artifactId>schema-iso19139.ca.HNAP</artifactId>
+                      <version>${gn.schemas.version>}</version>
+                      <type>jar</type>
+                      <overWrite>false</overWrite>
+                      <outputDirectory>${basedir}/src/main/webapp/WEB-INF/data/config/schema_plugins</outputDirectory>
+                      <includes>src/main/plugin</includes>
+                    </artifactItem>
+                  </artifactItems>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+   ```
 
+4. Use your profile, for example `mvn process-resources -Phnap`, to extract plugin from artifact during build.
+
+## Option: Fork and add a plugin to the build
```

Comment:
Is this option the same as using the "add-schema.sh"?
If so then the syntax for add-schema.sh should probably be added to this section.  It seems that the instructions to use this option should be consistent between this document and the schema plugin documentation.  The hnap module indicates to use add-schema.sh but this documentation indicates to update the file manually.

We could also add a similar script for the first option that would work similarly.(future enhancement). Unless add-schema.sh is being depreciated?

Reply:
This replaces add-schema.sh

Reply:
I updated the README.md to mention `add-schema.sh` here.

Reply:
The `maven-ant-plugin` technique worked, It does not appear that `add-schema.sh` script is required or will function as written.

I could also not tell from reading the `add-schema.sh` if it added additional dependencies to `web/pom.xml` (but I did not look too closely).